test/Rx/Functional/Operator/ZipTest.php

```diff
@@ -0,0 +1,615 @@
+            onError(220, $error2)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([
+            onError(220, $error2)
+        ], $results->getMessages());
+    }
+
+    public function testZipSomeError()
+    {
+        $error = new \Exception();
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(215, 2),
+            onCompleted(230)
+        ]);
+        $e2 = $this->createHotObservable([
+            onNext(150, 1),
+            onError(220, $error)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([
+            onError(220, $error)
+        ], $results->getMessages());
+    }
+
+    public function testZipErrorSome()
+    {
+        $error = new \Exception();
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(215, 2),
+            onCompleted(230)
+        ]);
+        $e2 = $this->createHotObservable([
+            onNext(150, 1),
+            onError(220, $error)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([
+            onError(220, $error)
+        ], $results->getMessages());
+    }
+
+    public function testZipSomeDataAsymmetric1()
+    {
+        $results = [];
+        for ($i = 0; $i < 5; $i++) {
+            $results[] = onNext(205 + $i * 5, $i);
+        }
+
+        /** @var Recorded[] $msgs1 */
+        $msgs1 = $results;
+
+        $results = [];
+        for ($i = 0; $i < 10; $i++) {
+            $results[] = onNext(205 + $i * 8, $i);
+        }
+
+        /** @var Recorded[] $msgs2 */
+        $msgs2 = $results;
+
+        $len = min(count($msgs1), count($msgs2));
+
+        $e1 = $this->createHotObservable($msgs1);
+        $e2 = $this->createHotObservable($msgs2);
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertEquals($len, count($results->getMessages()));
+
+        for ($i = 0; $i < $len; $i++) {
+            $sum  = $msgs1[$i]->getValue()->getValue() + $msgs2[$i]->getValue()->getValue();
+            $time = max($msgs1[$i]->getTime(), $msgs2[$i]->getTime());
+            $this->assertEquals($sum, $results->getMessages()[$i]->getValue()->getValue());
+            $this->assertEquals($time, $results->getMessages()[$i]->getTime());
+        }
```

Comment:
This test code is is quite hard to read. Can you write out the messages explicitly instead? You can make `$msgs1` and `$msg2` a bit smaller (e.g. 3 and 5 elements).